diff --git a/9999-Allow-Dark-Mode-RFP.patch b/9999-Allow-Dark-Mode-RFP.patch
new file mode 100644
index 0000000..6d4ed45
--- /dev/null
+++ b/9999-Allow-Dark-Mode-RFP.patch
@@ -0,0 +1,15 @@
+diff --git a/dom/base/Document.cpp b/dom/base/Document.cpp
+index 797661b3d8..7f8236bbe5 100644
+--- a/dom/base/Document.cpp
++++ b/dom/base/Document.cpp
+@@ -17226,10 +17226,6 @@ void Document::RemoveToplevelLoadingDocument(Document* aDoc) {
+
+ StylePrefersColorScheme Document::PrefersColorScheme(
+     IgnoreRFP aIgnoreRFP) const {
+-  if (aIgnoreRFP == IgnoreRFP::No &&
+-      nsContentUtils::ShouldResistFingerprinting(this)) {
+-    return StylePrefersColorScheme::Light;
+-  }
+
+   if (auto* bc = GetBrowsingContext()) {
+     switch (bc->Top()->PrefersColorSchemeOverride()) {
diff --git a/PKGBUILD b/PKGBUILD
index 7eed99a..916cf1f 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -31,11 +31,14 @@ options=(!emptydirs !makeflags !strip)
 source=("hg+$_repo"
         "firefox-nightly.desktop"
         "0001-Use-remoting-name-for-GDK-application-names.patch"
-		"D90627.diff")
+		"D90627.diff"
+        "9999-Allow-Dark-Mode-RFP.patch")
+
 sha512sums=('SKIP'
             '3ab341ad51c1fded9740c70c3eeb160750c24a333634837b646a86c4a431979d51ab4ccc9a3f87b899fdb960695dc710e78433b09dbf72e62eeda417761030f7'
             'afb4a230b3e87cfb71687b3fe375c78463e02a6f7b1daa15bf6127f6414c6c29bf2d8df372b59b4df7f90fc8929582e8aab4e3db5e8b54b1817c96aad00d92ea'
-            '02480231cec469fba1e03f669f0a47954bf633f0487ed27d4e0f135d100d9ae023044ef659cc32915aa8acb8dbcbfa2a6a69c78a94700961f9fa80a18bd2f4eb')
+            '02480231cec469fba1e03f669f0a47954bf633f0487ed27d4e0f135d100d9ae023044ef659cc32915aa8acb8dbcbfa2a6a69c78a94700961f9fa80a18bd2f4eb'
+            'eca1f06d7cf90c26168d994b6b71dec41aa408e7ede0a153fdbfd75195520bdad844437e839ed972fc1602abf98c73d8477784f9d86b03e17d3a6dd66323bcad')
 validpgpkeys=('14F26682D0916CDD81E37B6D61B7B526D98F0353') # Mozilla Software Releases <release@mozilla.com>
 
 # Google API keys (see http://www.chromium.org/developers/how-tos/api-keys)
@@ -61,6 +64,7 @@ prepare() {
   
   # https://bugzilla.mozilla.org/show_bug.cgi?id=1530052
   patch -Np1 -i ../0001-Use-remoting-name-for-GDK-application-names.patch
+  patch -Np1 -i ../9999-Allow-Dark-Mode-RFP.patch
 
   # https://phabricator.services.mozilla.com/D90627
   #patch -Np1 -i ../D90627.diff
